client/src/components/LandingNavbar/LandingNavbar.tsx

```diff
@@ -1,42 +1,118 @@
+    href: '/',
+  },
+];
 
 export default function LandingNavbar(): JSX.Element {
   const classes = useStyles();
+  const { toolbar, mobileIcon, mobileToolbar, drawerContainer, logo, linkItem, navbarDesktop, links } =
+    authNavbarUseStyles();
   const { pathname } = useLocation();
-  return (
-    <div className={classes.root}>
-      <AppBar position="static" className={` ${pathname === '/' ? classes.mainPageNavbar : classes.otherPageNavbar}`}>
-        <Toolbar>
-          <div className={classes.logo}>
-            <Link to="/">
-              <img src={logo} alt="logo" />
-            </Link>
-          </div>
-          <Typography className={classes.sitterPage}>
-            <Link to="#" className={` ${pathname === '/' ? classes.sitterLink : classes.otherSitterLink}`}>
-              BECOME A SITTER
-            </Link>
-          </Typography>
-          <div className={classes.login}>
-            <Button
-              component={Link}
-              to="/login"
-              variant="outlined"
-              color="inherit"
-              className={` ${pathname === '/' ? classes.loginBtn : classes.otherLoginBtn}`}
-            >
-              LOGIN
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const isMobileView = useMediaQuery('(max-width:600px)');
+
+  const displayDesktop = (
+    <Toolbar className={toolbar}>
+      <Link to="/">
+        <img src={Logo} className={logo} alt="logo" />
+      </Link>
+      <div className={navbarDesktop}>
+        <MenuList className={links}>
+          <MenuItem className={classes.menuItem}>
+            <Typography className={classes.sitterPage}>
+              <Link to="#" className={` ${pathname === '/' ? classes.sitterLink : classes.otherSitterLink}`}>
+                BECOME A SITTER
+              </Link>
+            </Typography>
+          </MenuItem>
+          <MenuItem className={classes.menuItem}>
+            <div className={classes.login}>
+              <Button
+                component={Link}
+                to="/login"
+                variant="outlined"
+                color="inherit"
+                className={` ${pathname === '/' ? classes.loginBtn : classes.otherLoginBtn}`}
+              >
+                LOGIN
+              </Button>
+            </div>
+          </MenuItem>
+          <MenuItem className={classes.menuItem}>
+            <Button component={Link} to="/signup" variant="contained" color="primary" className={classes.signUpBtn}>
+              SIGN UP
             </Button>
-          </div>
-          <Button component={Link} to="/signup" variant="contained" color="secondary" className={classes.signUpBtn}>
-            SIGN UP
-          </Button>
-        </Toolbar>
-      </AppBar>
-    </div>
+          </MenuItem>
+        </MenuList>
+      </div>
+    </Toolbar>
+  );
+
+  const displayMobile = () => {
+    const handleDrawerOpen = () => setIsDrawerOpen(true);
+    const handleDrawerClose = () => setIsDrawerOpen(false);
```

Comment:
As mentioned below, this is being repeated in this file and can be used as one function expression that both can share.